errbot/__init__.py

```diff
@@ -73,25 +120,37 @@ def some_command(self, msg, args):
     were given to the command by the user.
     """
 
-    def decorate(func, hidden=False, name=None, split_args_with='', admin_only=False, historize=True, template=None):
-        if not hasattr(func, '_err_command'):  # don't override generated functions
-            setattr(func, '_err_command', True)
-            setattr(func, '_err_re_command', False)
-            setattr(func, '_err_command_hidden', hidden)
-            setattr(func, '_err_command_name', name or func.__name__)
-            setattr(func, '_err_command_split_args_with', split_args_with)
-            setattr(func, '_err_command_admin_only', admin_only)
-            setattr(func, '_err_command_historize', historize)
-            setattr(func, '_err_command_template', template)
-        return func
-
-    if len(args):
-        return decorate(args[0], **kwargs)
-    else:
-        return lambda func: decorate(func, **kwargs)
-
-
-def re_botcmd(*args, **kwargs):
+    if args:
```

Comment:
I'm trying to wrap my head around why this is needed, when this is needed. Is also wonder if we can't do...

``` python
if not args:
   args = args[0]

return _tag_botcmd(bla...)
```

instead, to avoid having to more or less duplicate these two calls.